src/server/application/ServiceApi.ts

```diff
@@ -0,0 +1,59 @@
+import * as express from 'express'
+import { ApolloServer } from 'apollo-server-express'
+import { gql, PubSubEngine } from 'apollo-server'
+import { Bundle, Contract, Events } from '../core'
+import { ContractController } from './ContractController'
+import { ContractRepository } from './lib/ContractRepository'
+
+export type Config = {
+  contractController: ReturnType<typeof ContractController>
+  contractRepository: ContractRepository
+  pubSubClient: PubSubEngine
+}
+
+export function ServiceApi ({ contractController, contractRepository, pubSubClient }: Config) {
+  const app = express()
+  const apolloServer = new ApolloServer({
+    typeDefs: gql`
+      type SigningRequest {
+        transaction: String!
+      }
+      type Contract {
+        engine: String!
+        contractAddress: String!
+      }
+      type Query {
+        contracts: [Contract]!
+      }
+      type Mutation {
+        loadContract(contractAddress: String!, bundleLocation: String!): Boolean
+      }
+      type Subscription {
+        transactionSigningRequest(publicKey: String!): SigningRequest
+      }
+    `,
+    resolvers: {
+      Query: {
+        async contracts () {
+          const contracts = await contractRepository.findAll()
+          return contracts.map(({ address, bundle }: { address: Contract['address'], bundle: Bundle }) => {
+            return { contractAddress: address, engine: bundle.meta.engine }
+          })
+        }
+      },
+      Mutation: {
+        loadContract (_: any, { contractAddress, bundleLocation }: { contractAddress: string, bundleLocation: string }) {
+          return contractController.load(contractAddress, bundleLocation)
+        }
+      },
+      Subscription: {
+        transactionSigningRequest: {
+          subscribe: (_: any, { publicKey }: { publicKey: string }) => pubSubClient.asyncIterator(`${Events.SIGNATURE_REQUIRED}.${publicKey}`)
```

Comment:
So you landed on leaving this in here for now?

That makes sense, particularly as we approach demo. Will be easy to move once things stabilise

Reply:
Yes, it's a separate body of work, and I wanted to maintain the integrity of the integration test